packages/mapper/src/mapper.ts

```diff
@@ -463,15 +564,28 @@ export default class Mapper {
    */
   async findOneAndReplace(collection, filter, replacement, options = {}): Promise<any> {
     const findOneAndReplaceOptions: any = { ...options };
-    this.messageBus.emit('method:findOneAndReplace', collection._name, filter);
+    const db = collection._database._name;
+    const coll = collection._name;
 
     if ('returnNewDocument' in findOneAndReplaceOptions) {
       findOneAndReplaceOptions.returnDocument = findOneAndReplaceOptions.returnNewDocument;
       delete findOneAndReplaceOptions.returnNewDocument;
     }
+
+    this.messageBus.emit(
+      'mongosh:api-call',
+      {
+        method: 'findOneAndReplace',
+        class: 'Collection',
+        db,
+        coll,
+        arguments: { filter, replacement, findOneAndReplaceOptions }
```

Comment:
do we want to log the replacement document, or just the filters/options?

Reply:
i think the rule should be, emit everything then we decide what to log in the logger.ts and what to send to segment in telemetry.ts or however that will be called.

Reply:
I think my concern is that it can be a _very_ large document, or in case of `insertMany`, thousands of documents someone is reading from another file, and I don't want to pollute.